include/net/ieee802154_radio.h

```diff
@@ -96,6 +96,14 @@ struct ieee802154_filter {
 /* @endcond */
 };
 
+struct ieee802154_key {
+	uint8_t *key_value;
+	uint32_t key_frame_counter;
+	bool frame_counter_per_key;
+	uint8_t key_id_mode;
+	uint8_t key_index;
```

Comment:
I believe we need here an additional field for indication if the key should be stored or removed.